### PR TITLE
Ported the XrefPrefixes HC which checks that GO, MGI and ZFIN dbprima…

### PR DESCRIPTION
**Pull request by thomasmaurel**
…ry_acc prefix match an expected pattern. I have extended this test to include many xrefs following an expected pattern for both Vert and non-vert. The check has spotted the wrong IDs reported in this ticket: ENSCORESW-3196. I believe that this extended Datacheck now makes the PredictedXrefs check redundant (https://github.com/Ensembl/ensembl-datacheck/pull/126)

### CODE REVIEW

File: lib/Bio/EnsEMBL/DataCheck/Checks/XrefPrefixes.pm [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'XrefPrefixes',
+  DESCRIPTION    => 'Check that xrefs have the correct prefix for their dbprimary_acc',
+  GROUPS         => ['compare_core', 'xref'],
```
Comment (james-monkeyshines) [inline lib/Bio/EnsEMBL/DataCheck/Checks/XrefPrefixes.pm]:
```suggestion
  GROUPS         => ['xref'],
```

File: lib/Bio/EnsEMBL/DataCheck/index.json [hunk 2]
```diff
@@ -1369,6 +1369,16 @@
       "name" : "VariationSubset",
       "package_name" : "Bio::EnsEMBL::DataCheck::Checks::VariationSubset"
    },
+   "XrefPrefixes" : {
+      "datacheck_type" : "critical",
+      "description" : "Check that xrefs have the correct prefix for their dbprimary_acc",
+      "groups" : [
+         "compare_core",
```
Comment (james-monkeyshines) [inline lib/Bio/EnsEMBL/DataCheck/index.json]:
```suggestion
```

File: lib/Bio/EnsEMBL/DataCheck/Checks/XrefPrefixes.pm (hunk 1 above)
Comment (thomasmaurel) [reply lib/Bio/EnsEMBL/DataCheck/Checks/XrefPrefixes.pm]: Thanks, fixed in commit.

File: lib/Bio/EnsEMBL/DataCheck/index.json (hunk 2 above)
Comment (thomasmaurel) [reply lib/Bio/EnsEMBL/DataCheck/index.json]: Thanks, fixed in commit.

**Pull request merged**